pkg/rbeconfigsgen/rbeconfigsgen.go

```diff
@@ -0,0 +1,930 @@
+	}
+	return generatedFile{
+		name:     "java/BUILD",
+		contents: buf.Bytes(),
+	}, nil
+}
+
+// processTempDir creates a local temporary working directory to store intermediate files.
+func processTempDir(o *Options) error {
+	if o.TempWorkDir != "" {
+		s, err := os.Stat(o.TempWorkDir)
+		if err != nil {
+			return fmt.Errorf("got %q specified as option TempWorkDir but the path doesn't exist: %w", o.TempWorkDir, err)
+		}
+		if !s.IsDir() {
+			return fmt.Errorf("got %q specified as option TempWorkDir but the path doesn't point to a directory", o.TempWorkDir)
+		}
+		return nil
+	}
+	dir, err := ioutil.TempDir("", "rbeconfigsgen_")
+	if err != nil {
+		return fmt.Errorf("failed to create a temporary local directory to write intermediate files: %w", err)
+	}
+	o.TempWorkDir = dir
+	return nil
+}
+
+// genConfigBuild generates the contents of a BUILD file with a toolchain target pointing to the
+// C++ toolchain related rules generated by Bazel and a default platforms target.
+func genConfigBuild(o *Options) (generatedFile, error) {
+	if o.PlatformParams.CppToolchainTarget != "" {
+		return generatedFile{}, fmt.Errorf("<internal error> C++ toolchain target was already set")
+	}
+	// Populate the C++ toolchain target if C++ config generation is enabled.
+	if o.GenCPPConfigs {
+		o.PlatformParams.CppToolchainTarget = "//cc:cc-compiler-k8"
+		if o.OutputConfigPath != "" {
+			o.PlatformParams.CppToolchainTarget = fmt.Sprintf("//%s/cc:cc-compiler-k8", path.Clean(o.OutputConfigPath))
+		}
+	} else {
+		log.Printf("Not generating a toolchain target to be used for the C++ Crosstool top because C++ config generation is disabled.")
+	}
+	buf := bytes.NewBuffer(nil)
+	log.Printf("Fully resolved platform params=%v", o.PlatformParams)
+	if err := platformsToolchainBuildTemplate.Execute(buf, o.PlatformParams); err != nil {
+		return generatedFile{}, fmt.Errorf("failed to generate platform BUILD file: %w", err)
+	}
+	return generatedFile{
+		name:     "config/BUILD",
+		contents: buf.Bytes(),
+	}, nil
+}
+
+// copyCppConfigsToTarball copies the C++ configs generated by Bazel from the local filesystem at
+// 'inTarPath' to the output tarball represented by `outTar`.
+func copyCppConfigsToTarball(inTarPath string, outTar *tar.Writer) error {
+	in, err := os.Open(inTarPath)
+	if err != nil {
+		return fmt.Errorf("unable to open input tarball %q for reading: %w", inTarPath, err)
+	}
+	defer in.Close()
+	inTar := tar.NewReader(in)
+	pathPrefix := "cc"
+
+	for {
+		h, err := inTar.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error while reading input tarball %q: %w", inTarPath, err)
+		}
+		switch h.Typeflag {
+		case tar.TypeDir:
+			break
+		case tar.TypeReg:
+			if strings.HasSuffix(h.Name, "WORKSPACE") {
+				break
+			}
+			outH := &tar.Header{
+				Name:    path.Join(pathPrefix, h.Name),
+				Size:    h.Size,
+				Mode:    h.Mode,
+				ModTime: time.Unix(0, 0),
```

Comment:
Delete, use the value from `inTar`, or the current time?  Unless you're setting it to something nonzero, this is redundant.  Ditto below.

Reply:
Done but I didn't get what you meant by `Ditto below`. The one below constructs a header from scratch and there's no input tarball to initialize a header from.

Reply:
I meant the `time.Unix(0, 0)`, in particular (sorry, still getting used to github reviews, meant to focus on just that bit, not the whole header).  Any reason NOT to use `time.Now()` in both places?

Reply:
Ah I see. As I commented in the code here, I noticed some weirdness with the mod times of the input tarball so I just decided to use epoch for everything to avoid the weirdness and also make it deterministic. So going to keep this for now.